src/transformStateWithClones.js

```diff
@@ -65,7 +65,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const result = [];
+  let currentTransform = Object.assign({}, state);
+
+  for (const transform in transforms) {
+    switch (transforms[transform].operation) {
+      case 'addProperties':
+        Object.assign(currentTransform, transforms[transform].properties);
+        break;
+
+      case 'removeProperties':
+        for (const key of transforms[transform].properties) {
+          delete currentTransform[key];
+        }
+        break;
+
+      case 'clear':
+        currentTransform = {};
+        break;
+
+      default:
+        break;
```

Comment:
no need to write break after default and you can throw an error here, or just show in the console that something wrong